packages/neos-ui-backend-connector/src/index.ts

```diff
@@ -32,6 +32,7 @@ export const initializeJsAPI = (parent: {[propName: string]: any}, {alias = 'neo
     addLibrary('use', initializeUse(addLibrary, neos));
     addLibrary('q', initializeFlowQuery(routes));
     addLibrary('endpoints', initializeEndpoints(routes));
+    addLibrary('routes', routes);
```

Comment:
i dont think exposing this and possibly increasing our surface of the experimental extensibility api makes sense.
I can only assume that the routes were explicitly no part of `window` and that we should always `yield select()` the state.

Reply:
Hmm, ok but declaring the `defaultModule` as endpoint also sounds bad and something like `$get('routes.core.modules.defaultModule', neos)` is not working in the saga. So I need to check if I can use it somehow in the saga.

Reply:
hmm i see the `routes` are indeed directly passed to `<Root />` which creates a react context but we cant use those in redux.

In that case i would say we either have to store the value in redux which makes no sense right? Or we store it in a value in a module. And refactor the code later. That will make the code smell obvious without using `window` and exposing too much.

Reply:
Stupid question: Would you save all routes to the state, or just the single one? Think about bringing the routes to the system state. So we would have next to the system.authenticationTimeout also system.routes.

Just would ask for the intention before building it and then reverting it again ;)

Reply:
Il have to ask wilhelm, to hard the question is

Reply:
he solved the puzzle :D We can add the `routes` via sage middleware run as initial parameter to each saga.

SEE https://github.com/neos/neos-ui/pull/3713/commits/bf48d911d5076c58199b8865a93ef713e6faecf1